### ISSUE TITLE
Why prox_weight_nw divide  depth_weights?

### ISSUE THREAD

**Issue by CJCHEN1230** [NagabhushanSN95/Pose-Warping#7]
Thanks for sharing your code. I trace the code and found something I can't figure it out.
How do you normalize the _depth_weights_?  What does 50 mean?
```
sat_depth1 = torch.clamp(depth1, min=0, max=1000)
log_depth1 = torch.log(1 + sat_depth1)  # (b, 2, h, w)
depth_weights = torch.exp(log_depth1 / log_depth1.max() * 50)
```
And why every weight should divide the `depth_weight` ?

````
weight_nw = torch.moveaxis(prox_weight_nw * mask1 * flow12_mask / depth_weights, [0, 1, 2, 3], [0, 3, 1, 2])
weight_sw = torch.moveaxis(prox_weight_sw * mask1 * flow12_mask / depth_weights, [0, 1, 2, 3], [0, 3, 1, 2])
weight_ne = torch.moveaxis(prox_weight_ne * mask1 * flow12_mask / depth_weights, [0, 1, 2, 3], [0, 3, 1, 2])
weight_se = torch.moveaxis(prox_weight_se * mask1 * flow12_mask / depth_weights, [0, 1, 2, 3], [0, 3, 1, 2])
````

**Comment by NagabhushanSN95** [comment]
Hi, 

1. 50 is just a hyper-parameter that worked well. I vaguely remember it being connected to division by 2, but I don't remember exactly what it is.
2. division by `depth_weight` is similar to keeping a z-buffer and picking the nearest point. Basically, dividing with `depth_weights` makes sure that the nearest point has a higher weight than a farther point when both the points map to the same pixel in the target view.

**Comment by CJCHEN1230** [comment]
Thanks for your explanation. Really appreciate!
